tests/unit/test.index.js

```diff
@@ -0,0 +1,313 @@
+      logStream.on('data', function(data) {
+        dataReceived.push(data);
+      });
+
+      // Send the events
+      if (useLogs) {
+        mockClient._eventHandlers['console-api-call'](logEvent);
+      }
+      if (useErrors) {
+        mockClient._eventHandlers['page-error'](errorEvent);
+      }
+
+      // Return a promise that yields to the event loop
+      return new Promise(function(resolve, reject) {
+        setImmediate(resolve);
+      });
+
+    }).then(function(logStream) {
+
+      if (useLogs) {
+        test.equal(dataReceived.shift(), logEvent);
+      }
+      if (useErrors) {
+        test.equal(dataReceived.shift(), errorEvent);
+      }
+
+      test.ok(mockClient._mocked.assert());
+      return test.done();
+
+    }).catch(function(err) {
+
+      console.error(err);
+      test.done();
+
+    });
+
+  };
+
+}
+
+var APP_ID = '8675309';
+
+var APP_NAME = 'Test App';
+
+var APP_MANIFEST_URL = 'app://' + APP_ID + '/manifest.webapp';
+
+var MANIFEST = {
+  id: APP_ID,
+  manifestURL: APP_MANIFEST_URL,
+  name: APP_NAME
+};
+
+var APP_ID_NOT_RUNNING = 'ou812';
+
+var MANIFEST_NOT_RUNNING = {
+  id: APP_ID_NOT_RUNNING,
+  manifestURL: 'app://' + APP_ID_NOT_RUNNING + '/manifest.webapp',
+  name: 'Not running app'
+};
+
+var MANIFEST_NOT_INSTALLED = {
+  id: 'lolidk',
+  manifestURL: 'app://lolidk/manifest.webapp',
+  name: 'Not installed app'
+};
+
+var INSTALLED_APPS = [
+  MANIFEST,
+  MANIFEST_NOT_RUNNING
+];
+
+var RUNNING_APPS = [
+  'app://system.gaiamobile.org/manifest.webapp',
+  'app://costcontrol.gaiamobile.org/manifest.webapp',
+  'app://callscreen.gaiamobile.org/manifest.webapp',
+  'app://verticalhome.gaiamobile.org/manifest.webapp',
+  APP_MANIFEST_URL,
+  'app://findmydevice.gaiamobile.org/manifest.webapp',
+  'app://settings.gaiamobile.org/manifest.webapp',
+  'app://ad97abe0-00bf-11e5-ab12-d5ad5262fbe2/manifest.webapp',
+  'app://keyboard.gaiamobile.org/manifest.webapp'
+];
+
+// nodemock expects an empty function to indicate a callback parameter
+var CALLBACK_TYPE = function() {};
+
+var CONSOLE_METHOD_NAMES = [
+  'startListening', 'stopListening', 'getCachedLogs',
+  'clearCachedLogs', 'evaluateJS'
+];
+
+var JS_EXPR = '1 + 1';
+
+function buildMockClient(options) {
+
+  var mocked = nodemock
+    .mock('getInstalledApps')
+    .takes(CALLBACK_TYPE)
+    .calls(0, [null, INSTALLED_APPS]);
+
+  mocked.mock('listRunningApps')
+    .takes(CALLBACK_TYPE)
+    .calls(0, [null, RUNNING_APPS]);
+
+  var consoleMethodNames = options.consoleMethodNames || CONSOLE_METHOD_NAMES;
+
+  if (!options.shouldCallConsoleMethods) {
+    consoleMethodNames.forEach(function(name) {
+      mocked.mock(name).fail();
+    });
+  } else {
+
+    if (consoleMethodNames.indexOf('evaluateJS') !== -1) {
+      mocked.mock('evaluateJS')
+        .takes(JS_EXPR, CALLBACK_TYPE)
+        .calls(1, [null, true]);
+    }
+
+    consoleMethodNames.forEach(function(name) {
+      if (name === 'evaluateJS') { return; }
```

Comment:
I like using that because it's a compact expression for a simple test, but I'm not super attached to it